### ISSUE TITLE
Documentation: Cookie Policy

### ISSUE THREAD

**Issue by mbecker** [mikecao/umami#15]
Hi,

thanks for sharing this great project! Looks awesome!

However for the cookie policy we have to add the name of cookies the scripts set on a page. I've noticed the following cookie from unami on the page I've added the tracking javascript:

```sh
umami.auth
```

Does the script set any other cookie?

Unfortunately I do not find the max-age/expiration date for that cookie. May I ask you how long is the cookie stored?

From what I see I would say the following:

```sh
Cookie name: unami.auth
Max age: 1 day
```

Maybe a feature request: Add the option to rename that cookie to have a more custom way of using unami.

**Comment by mikecao** [comment]
The tracking script doesn't set any cookies. `umami.auth` is the login cookie for your umami application. You're probably seeing it on the tracked page because it points to the domain where your umami is installed.

**Comment by mbecker** [comment]
Hi @mikecao , thnaks for clarification!

Indeed, visiting the page with the tracking script in incognito modus doesn't show the cookie. Great!

May I ask you how do you count unique visitors?

**Comment by nh2** [comment]
> May I ask you how do you count unique visitors?

@mbecker I also had the same question; just found it's explained here:

* https://github.com/mikecao/umami/issues/27#issuecomment-677344693
* https://github.com/mikecao/umami/issues/155#issuecomment-702391188

**Comment by bernhardh** [comment]
Not sure if I am right, but is this [discussion](https://news.ycombinator.com/item?id=24201595) with @mikecao ? There the user `` said:

> Maybe that's not fully compliant, I don't know, so I went ahead and removed any mention of GDPR from the website.

But the GDPR info is still in the docs - https://umami.is/docs/faq

So is umami GDPR compliant?

**Comment by mikecao** [comment]
@bernhardh A lot of the code has changed since umami was first released. Initially it used localStorage by default and that could be interpreted by some as using a cookie. Now the default behavior is to not store anything, so to the best of my knowledge it is GDPR compliant.

**Comment by david-mears-2** [comment]
Developers in the UK should note that, at least at the time of writing,

> You must obtain consent for:  [...] tracking or profiling individual visitors or categories of visitors (eg based on their IP address or the pages they visited on your website).

https://ico.org.uk/for-organisations/direct-marketing-and-privacy-and-electronic-communications/guidance-on-the-use-of-storage-and-access-technologies/what-are-the-exceptions/#statistical

Since umami fingerprints users by IP + device + browser and only cycles the id every month (as per docs https://umami.is/docs/metric-definitions#location-metrics and implemented [here](https://github.com/umami-software/umami/blob/2260520ae58b7b79327f0f50e4d465931b07f41c/src/app/api/send/route.ts#L133)), it would count as tracking individual visitors. Indeed, umami's docs [talk about tracking the number of visitors](https://umami.is/docs/metric-definitions#location-metrics) in addition to visits, and advertise a 'retention' report feature, so we're definitely in the realm of "tracking or profiling individual visitors". Comparable analytics solutions use shorter salt rotation periods in order to avoid tracking individual visitors: e.g. 24 hours for Fathom, 24 hours for Posthog, 8 hours for goatcounter. These are more fairly described as tracking _sessions_ or _visits_ as opposed to visitors.

There is currently no way to disable this tracking of unique visitors (see https://github.com/umami-software/umami/issues/4031, https://github.com/umami-software/umami/issues/4033). So user consent would be required to use umami in the UK jurisdiction.

Umami is only UK GDPR compliant if your site obtains user consent before running any umami scripts.

**Comment by whysthatso** [comment]
ianal but i have talked to some about the issues, and of course it's complicated. 

I'm running a selfhosted Umami instance for audience measurement and as a load indicatore, so no tracking intended.
however, i tend to follow @david-mears-2 argument to a degree.

Generally i understand that stating strong, absolute propositions like:

>Yes, Umami does not collect any personally identifiable information and anonymizes all data collected. Users cannot be identified and are never tracked across websites.

is probably insufficient to fully dismiss consent requirements. As i further understand, a strong case for compliance can be made when these attributes are fulfilled:

- self-hosted Umami,
- EU hosting,
- first-party only,
- no advertising,
- short retention,
- anonymized IPs,
- no cross-site tracking,
- transparent privacy notice,
- optional opt-out,
- analytics only

I think the only lacking features are an automated retention policy setting, and the optional opt-out.

The argument goes that the user does not benefit from the analytics, so their intent when using a site is not analytics. It's for the site operator. It is also not strictly technically necessary, like a webserver keeping track of an open session with a unique user (which does not need consent, but is implied by the technical necessity)

I understand also, that the regulations are not focused on cookie or not; it's more generally aimed at locally available storage, which would include other storage mechanisms from the browser. But this might all be old news, because the code might have changed since this comment: https://github.com/umami-software/umami/issues/27#issuecomment-677344693

It seems to me that a "cookie banner" aka opt-in is indeed not needed given the above list of requirements is fullfilled, but an opt out checkbox that can be integrated in the site's imprint, etc. like https://matomo.org/faq/general/faq_20000/#customise-and-publish-the-matomo-opt-out-form-html does it

So in short, i think the documentation should possibly be toned down a bit, and i do also support the two features:

- retention policies #1277
- opt-out integration #27

I'm just going to reference the other issues i found here, for the future: #76 #2929 #27 #155 #1277

@mikecao what do you think?
